model_compression_toolkit/common/quantization/quantization_params_generation/symmetric_selection.py

```diff
@@ -0,0 +1,190 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import numpy as np
+from scipy import optimize
+
+import model_compression_toolkit.common.quantization.quantization_config as qc
+from model_compression_toolkit.common.constants import MIN_THRESHOLD, THRESHOLD
+from model_compression_toolkit.common.quantization.quantization_params_generation.kl_selection import \
+    _kl_error_histogram, _kl_error_function
+from model_compression_toolkit.common.quantization.quantization_params_generation.lp_selection import \
+    _lp_error_histogram
+from model_compression_toolkit.common.quantization.quantization_params_generation.mae_selection import \
+    _mae_error_histogram
+from model_compression_toolkit.common.quantization.quantization_params_generation.mse_selection import \
+    _mse_error_histogram
+from model_compression_toolkit.common.quantization.quantization_params_generation.qparams_search import \
+    kl_symmetric_qparams_selection_histogram_search_error_function, \
+    qparams_histogram_minimization, qparams_tensor_minimization, symmetric_qparams_selection_per_channel_search
+from model_compression_toolkit.common.quantization.quantizers.quantizers_helpers import get_tensor_max, quantize_tensor
+
+from model_compression_toolkit.common.similarity_analyzer import compute_mse, compute_mae, compute_lp_norm
+
+
+def symmetric_selection_tensor(tensor_data: np.ndarray,
+                               p: int,
+                               n_bits: int,
+                               per_channel: bool = False,
+                               channel_axis: int = 1,
+                               n_iter: int = 10,
+                               min_threshold: float = MIN_THRESHOLD,
+                               quant_error_method: qc.QuantizationErrorMethod = qc.QuantizationErrorMethod.MSE) -> dict:
+    """
+    Compute the optimal threshold based on the provided QuantizationErrorMethod to quantize the tensor.
+    Different search is applied, depends on the value of the selected QuantizationErrorMethod.
+
+    Args:
+        tensor_data: Tensor content as Numpy array.
+        p: p-norm to use for the Lp-norm distance.
+        n_bits: Number of bits to quantize the tensor.
+        per_channel: Whether the quantization should be per-channel or not.
+        channel_axis: Output channel index.
+        n_iter: Number of iterations to search for the optimal threshold (not used for this method).
+        min_threshold: Minimal threshold to use if threshold is too small (not used for this method).
+        quant_error_method: an error function to optimize the parameters' selection accordingly.
+
+    Returns:
+        Optimal threshold to quantize the tensor in a symmetric manner.
+    """
+
+    signed = np.any(tensor_data < 0)  # check if tensor is singed
+    unsigned_tensor_data = np.abs(tensor_data)
```

Comment:
Maybe we shouldn't check the sign, as it's always signed in weights quantization

Reply:
I followed the same technique that is used for power-of-two threshold selection, you can see in the method [qparams_selection_tensor_search](https://github.com/ofirgo/model_optimization/blob/6c09e77ee9c17a04577d4162b87d5d5ae6420bec/model_compression_toolkit/common/quantization/quantization_params_generation/qparams_search.py#L55) that it also checks signed value and passes it to the [quantize_tensor](https://github.com/ofirgo/model_optimization/blob/6c09e77ee9c17a04577d4162b87d5d5ae6420bec/model_compression_toolkit/common/quantization/quantization_params_generation/qparams_search.py#L84) method.
Is it a bug? if so, I'll change this is my implementation and in the original.

Reply:
Yeah... I'm afraid it's a bug.
@haihabi What do you think? We should pass signed=True always in the case of weights quantization, right?

Reply:
Resolved offline